staging/src/k8s.io/apimachinery/pkg/labels/selector.go

```
@@ -358,6 +364,22 @@ func (lsel internalSelector) String() string {
 	return strings.Join(reqs, ",")
 }
 
+// RequiresExactMatch introspect whether a given selector requires a single specific field
+// to be set, and if so returns the value it requires.
+func (lsel internalSelector) RequiresExactMatch(label string) (value string, found bool) {
+	for ix := range lsel {
+		if lsel[ix].key == label {
+			switch lsel[ix].operator {
+			case selection.Equals, selection.DoubleEquals, selection.In:
+				if len(lsel[ix].strValues) == 1 {
+					return lsel[ix].strValues[0], true
+				}
```

return "", false
otherwise (fail fast)